src/transformers/models/siglip/modeling_siglip.py

```diff
@@ -926,8 +926,8 @@ def __init__(self, config: SiglipTextConfig):
         self.embeddings = SiglipTextEmbeddings(config)
         self.encoder = SiglipEncoder(config)
         self.final_layer_norm = nn.LayerNorm(embed_dim, eps=config.layer_norm_eps)
-        #if not (config.vision_config.image_size==256 and config.text_config.vocab_size==250000 and config.vision_config.patch_size==16):
-        self.head = nn.Linear(embed_dim, embed_dim)
+        if not hasattr(self.config, "no_head"):
```

Comment:
This should be a boolean attribute, which is cleaner to use that `hasattr`.

I don't see updates to configuration_siglip.py, the no_head argument should be added there

Reply:
addressed